### PR TITLE
fix: (Platform) add general fixes found during MetaUI implementation

### PR DESCRIPTION
**Pull request by fkolar**
#### Please provide a link to the associated issue.
1 is fixing #3999 and and other ones are found during implementations

#### Please provide a brief summary of this pull request.

- missing exports
- cleaning up code which is not needed
- Introducing interface method needed by FormField implementaion

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by dimamarksman — APPROVED** [review]

File: libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts [hunk 1]
```diff
@@ -69,19 +69,15 @@ export class FormFieldComponent implements FormField, AfterContentInit, AfterVie
     @Input()
     noLabelLayout = false;
 
+    @Input()
+    validators: Array<ValidatorFn> = [Validators.nullValidator];
+
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts:74]: In general all these Inputs should have typedocs as that's how the API is built in the documentation.

File: libs/platform/src/lib/components/form/form-group/form-group.component.ts [hunk 2]
```diff
@@ -334,6 +335,13 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
         this.formFieldChildren.forEach((formField) => this.updateFormFieldProperties(formField));
     }
 
+    /** @hidden */
+    private validateErrorHandler(): void {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/form-group/form-group.component.ts:339]: should be prefixed with `_`

File: libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts [hunk 3]
```diff
@@ -97,6 +93,9 @@ export class FormFieldComponent implements FormField, AfterContentInit, AfterVie
     @Input()
     i18Strings: TemplateRef<any>;
 
+    @Input()
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts:96]: same comment for the /** */

**Review by Betrozov — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]
LGTM

File: libs/platform/src/lib/components/form/form-group/form-group.component.ts (hunk 2 above)
Comment (fkolar) [reply libs/platform/src/lib/components/form/form-group/form-group.component.ts:339]: Please see https://github.com/SAP/fundamental-ngx/issues/4248

File: libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts (hunk 3 above)
Comment (fkolar) [reply libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts:96]: https://github.com/SAP/fundamental-ngx/issues/4247

**Review by InnaAtanasova — APPROVED** [review]

**Comment by fkolar** [comment]
Closing as these fixes are outdated and are part of other PRs

**Pull request closed without merging**